onset_maproom/calc.py

```diff
@@ -61,13 +102,50 @@ def onset_date(
     dry_spell_search,
     time_coord="T",
 ):
-    """Finds the first wet spell of wet_spell_length days
-    where cumulative rain exceeds wet_spell_thresh,
-    with at least min_wet_days count of wet days (greater than wet_thresh),
-    not followed by a dry spell of dry_spell_length days of dry days (not wet),
-    for the following dry_spell_search days
-    returns the time delta rom the first day of daily_rain
-    to the first wet day in that wet spell
+    """ Calculate onset date.
+
+        Finds the first wet spell of 'wet_spell_length' days where: 
+            Cumulative rain exceeds 'wet_spell_thresh',
+            With at least 'min_wet_days' count of wet days (greater than 'wet_thresh'),
+            Not followed by a dry spell of 'dry_spell_length' days of dry days (not wet),
+            For the following 'dry_spell_search' days
+    
+        Parameters
+        ----------
+        daily_rain : DataArray
+                    Array of daily rainfall values.
+        wet_thresh : int
+                    Rainfall threshold to determine wet day.
+        wet_spell_length : int
+                    Length in days of running window when 'wet_thresh' is to be met to define a wet spell.
+        wet_spell_thresh : int
+                    Threshold of rainfall to be reached during 'wet_spell_length'
+                    window to define a wet spell.
+        min_wet_days : int
+                    Minimum number of wet days in 'wet_spell_length' window when it rained at or above
+                    'wet_spell_thresh' to be considered a wet spell.
+        dry_spell_length : int
+                    Length in days of dry spell that would invalidate a wet spell as onset date
+                    if found within the 'dry_spell_search' days following the 'wet_spell_length'-day window
+                    that met the wet spell criteria.
+        dry_spell_search : int
+                    Length in days to search for a 'dry_spell_length'-day dry spell after a wet spell
+                    is found that would invalidate the wet spell as onset date.
+        time_coord : str, optional
+                    Time grid in 'soil_moisture' (the default is "T", which implies
+                    naming convention of time grid as "T" in 'soil_moisture')         
+        Returns
+        -------
+        onset_delta : DataArray    
```

Comment:
precise data type is TimeDelta?

Reply:
I put this because the output is a dataArray with a single value in it. So should I just put timeDelta as the data type?

Reply:
`DataArray[np.timedelta64]`

BTW I presume it's not really a single value, but a single value per X, Y point.

Reply:
X, Y, or whatever else that is not time_coord

Reply:
So then in a case like this you would say the dataType is the type of value within the array and not that the output is the array itself?

Reply:
> So then in a case like this you would say the dataType is the type of value within the array and not that the output is the array itself?

No, not at all. `DataArray` is correct; but `DataArray[np.timedelta64]` is also correct, and more precise.